### PR TITLE
GitHub actions

### LINKED ISSUE #52: Setup GitHub Actions
**Description**
GitHub Actions is providing a feasible alternative to CircleCI for running tests on branches and PRs. It has been implemented in Vocabulary creativecommons/vocabulary#245 and along similar lines must also be implemented in this repository.

**Problems solved**
GitHub Actions has similar features, better integration and higher usage limits as compared to CircleCI. Switching would be the best option.

### PR DESCRIPTION
**Pull request by kushaggarwal**
## Fixes

Fixes #52 by @dhruvkb 

## Description

This PR aims at the setup of github actions for fonts library

## Technical details

The implementation of the same follows the jobs and steps associated with the current CircleCI config

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kushaggarwal** [comment]
Hello @dhruvkb as in the CircleCI config currently exist only the job for lint and build taking the reference from the vocabulary github action I have added the job for unit and snapshot test 
Request your review and changes / improvements for the same 
Thank You

**Comment by dhruvkb** [comment]
@kushaggarwal I appreciate you adding the unit and snapshot tests. The only concern I have is that the snapshot test is written incorrectly and will certainly fail.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      uses: actions/cache@v1
+      with:
+        path: ~/.npm  # npm caches files in ~/.npm
+        key: node-${{ hashFiles('**/package-lock.json') }}
+
+    - name: Install dependencies
+      run: npm install --prefer-offline --no-audit
+
+    - name: Run tests
+      run: npm run test:snapshot
```
Comment (dhruvkb) [inline .github/workflows/ci.yml]: This will not work.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
I am pretty sure this will not work. Can you add that in another PR?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      uses: actions/cache@v1
+      with:
+        path: ~/.npm  # npm caches files in ~/.npm
+        key: node-${{ hashFiles('**/package-lock.json') }}
+
+    - name: Install dependencies
+      run: npm install --prefer-offline --no-audit
+
+    - name: Run tests
+      run: npm run test:
```
Comment (dhruvkb) [inline .github/workflows/ci.yml]: There's also a syntax error on this line

Comment (dhruvkb) [reply .github/workflows/ci.yml]: Keep track of the status in your [Actions tab](https://github.com/kushaggarwal/fonts/actions).

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kushaggarwal) [reply .github/workflows/ci.yml]: Okay @dhruvkb will do so

File: .github/workflows/ci.yml (hunk 2 above)
Comment (kushaggarwal) [reply .github/workflows/ci.yml]: Okay Thank you

**Comment by dhruvkb** [comment]
@kushaggarwal the build is still failing, see https://github.com/kushaggarwal/fonts/actions.

**Comment by dhruvkb** [comment]
Since there are no running tests, I'm gonna take a risk and merge it.

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**